### PR TITLE
Scenery tool down

### PR DESCRIPTION
**Pull request by duncanspumpkin**
This isn't finished but I thought i better put it up for visibility. This should eventually complement @zsilencer 's #1009 and @Garogolun viewport interaction code. 

I've changed `map_element.properties.scenerymultiple.type` to a uint16 and this will probably cause issues with anyone's work relating to large scenery items but its a required change since it should be a uint16. 

GAME_COMMAND_52 -> 55 are all commands for changing the colour of scenery probably quite simple functions but I haven't had a look at them.

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
Ready for merging. The whole structure of this code could do with a good bit of refactoring but I would prefer to do that after its confirmed no bugs have been made.

File: src/windows/top_toolbar.c [hunk 1]
```diff
@@ -626,70 +628,801 @@ static void window_top_toolbar_paint()
...
+		RCT2_GLOBAL(RCT2_ADDRESS_GAME_COMMAND_ERROR_STRING_ID, rct_string_id) = 3103;
+		game_do_command(
+			grid_x,
+			1 | (map_element->type << 8),
+			grid_y,
+			map_element->base_height | (map_element->properties.scenery.type << 8),
+			GAME_COMMAND_52,
+			0,
+			window_scenery_primary_colour | (window_scenery_secondary_colour << 8));
+	}
```
Comment (Garogolun) [inline src/windows/top_toolbar.c]: Don't you need `break`s here?

Comment (duncanspumpkin) [inline src/windows/top_toolbar.c]: Yeah woops

File: src/windows/top_toolbar.c [hunk 2]
```diff
@@ -626,70 +628,801 @@ static void window_top_toolbar_paint()
...
+		}
+
+		uint8 rotation = window_scenery_rotation;
+		rotation -= RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_ROTATION, uint8);
+		rotation &= 0x3;
+		
+		sint16 z = map_element->base_height;
+
+		if (map_element->properties.path.type & (1 << 2)){
+			if (rotation != ((map_element->properties.path.type & 3) ^ 2)){
```
Comment (Garogolun) [inline src/windows/top_toolbar.c]: I'm probably not the first to have figured this out, but: the third bit indicates whether the path is sloped and the first two bits contain the direction when walking upwards. Xor-ing a direction with 2 seems to yield the opposite direction.

**Pull request merged**